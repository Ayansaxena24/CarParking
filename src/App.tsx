import SlotPage from './SlotPage'

import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import ExitCar from './ExitCar'

const App = () => {
  return (
    <Router>
     <Routes>
      <Route path="/" element={<SlotPage />} />
      <Route path="/vacate/:id" element={<ExitCar />} />
      </Routes> 
    </Router>
  )
}

export default App
