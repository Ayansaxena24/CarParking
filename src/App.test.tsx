import { render } from '@testing-library/react';
import { Provider } from 'react-redux'; // Import the correct Provider component
// import { BrowserRouter as Router } from 'react-router-dom';
import store from './redux/Store'; // Import your Redux store
import App from './App';

test('renders learn react link', () => {
    const wrapper = render(
        <Provider store={store}> {/* Pass the store as a prop to the Provider component */}
            {/* <Router> */}
                <App />
            {/* </Router> */}
        </Provider>
    );

    expect(wrapper).toMatchSnapshot();
});
