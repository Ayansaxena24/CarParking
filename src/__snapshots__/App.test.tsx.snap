// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <div
        class="w-full h-full "
      >
        <div>
          <div
            class="w-full flex justify-center items-center py-2 bg-[conic-gradient(at_top,_var(--tw-gradient-stops))] from-gray-700 via-gray-900 to-black relative"
          >
            <p
              class=" text-3xl font-bold text-white "
            >
              PARKING
            </p>
            <p
              class="absolute right-8 text-white hover:text-violet-400 duration-300 ease-in-out "
            >
               About Us 
            </p>
          </div>
        </div>
        <div>
          <hr
            class="text-violet-500 py-0.2"
          />
        </div>
        <div
          class="bg-gradient-to-b from-gray-900 to-gray-600 bg-gradient-to-r mr-10 w-screen h-screen"
        >
          <div
            class="pt-4 mr-10 flex justify-end"
          >
            <p
              class="text-lg font-semibold text-white"
            >
              Welcome, Owner
            </p>
          </div>
          <div
            class="flex flex-col justify-center items-center absolute top-[50%] right-[30%] border-2 rounded-md px-10 py-6"
          >
            <div
              class="grid grid-cols-2 space-x-2"
            >
              <p
                class="text-white mt-1"
              >
                Enter required number of Parking Spaces
              </p>
              <input
                class="border-2 mb-2 border-gray-400 bg-gray-200 rounded-md p-1"
                type="text"
                value="1"
              />
            </div>
            <div
              class="mt-4"
            >
              <button
                class="text-white rounded-lg border-2 px-2 py-1 hover:shadow-lg hover:shadow-violet-400 duration-300 ease-in-out hover:border-violet-400 "
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
     
    <div
      class="w-full h-full "
    >
      <div>
        <div
          class="w-full flex justify-center items-center py-2 bg-[conic-gradient(at_top,_var(--tw-gradient-stops))] from-gray-700 via-gray-900 to-black relative"
        >
          <p
            class=" text-3xl font-bold text-white "
          >
            PARKING
          </p>
          <p
            class="absolute right-8 text-white hover:text-violet-400 duration-300 ease-in-out "
          >
             About Us 
          </p>
        </div>
      </div>
      <div>
        <hr
          class="text-violet-500 py-0.2"
        />
      </div>
      <div
        class="bg-gradient-to-b from-gray-900 to-gray-600 bg-gradient-to-r mr-10 w-screen h-screen"
      >
        <div
          class="pt-4 mr-10 flex justify-end"
        >
          <p
            class="text-lg font-semibold text-white"
          >
            Welcome, Owner
          </p>
        </div>
        <div
          class="flex flex-col justify-center items-center absolute top-[50%] right-[30%] border-2 rounded-md px-10 py-6"
        >
          <div
            class="grid grid-cols-2 space-x-2"
          >
            <p
              class="text-white mt-1"
            >
              Enter required number of Parking Spaces
            </p>
            <input
              class="border-2 mb-2 border-gray-400 bg-gray-200 rounded-md p-1"
              type="text"
              value="1"
            />
          </div>
          <div
            class="mt-4"
          >
            <button
              class="text-white rounded-lg border-2 px-2 py-1 hover:shadow-lg hover:shadow-violet-400 duration-300 ease-in-out hover:border-violet-400 "
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
